api/src/poem_backend/events.rs

```diff
@@ -87,14 +87,17 @@ impl EventsApi {
         event_key: EventKey,
     ) -> BasicResultWith404<Vec<Event>> {
         let latest_ledger_info = self.context.get_latest_ledger_info_poem()?;
+        let ledger_version = latest_ledger_info.version();
+        let last_sequence_number = self
```

Comment:
why don't we utilize the DB interface to get the latest events?

https://github.com/aptos-labs/aptos-core/blob/c817048580bcc422a268591bd214ee66ab9dcbb1/storage/aptosdb/src/lib.rs#L1074
https://github.com/aptos-labs/aptos-core/blob/c817048580bcc422a268591bd214ee66ab9dcbb1/storage/aptosdb/src/lib.rs#L647

Reply:
@msmouse :

current get_events API doesn't accept ledger_version, but we need API to be consistent about what it returns. That means that if new events are inserted from fetching latest_ledger_info to the time calling get_events, paging would be misaligned, and I would need to have more complicated logic, and issue multiple get_events calls.

So I have two options - adding ledger_version to the get_events call, or adding this new function to return last sequence number.
API is similar between get_transactions and get_events, but storage functions are not aligned. So I wanted to pick a solution that makes the API logic as similar between them as possible. Given transactions "sequence numbers" are versions, and storage provides that API to fetch it, adding API to fetch latest sequence number for event made the usage look very similar, and allowing me to reuse code (i.e. page.compute_start method I added).

Let me know if you have a different suggestion, that solves the first problem, and has consistency between the APIs?

Reply:
https://github.com/aptos-labs/aptos-core/pull/2652 let's see if this passes the tests.